packages/front-end/components/Settings/UpgradeModal/index.tsx

```diff
@@ -45,7 +55,22 @@ export default function UpgradeModal({ close, source }: Props) {
     false
   );
 
-  const { name, email, accountPlan, license, effectiveAccountPlan } = useUser();
+  const {
+    name,
+    email,
+    accountPlan,
+    license,
+    effectiveAccountPlan,
+    commercialFeatureLowestPlan,
+    seatsInUse,
+  } = useUser();
+  // These are some Upgrade CTAs throughout the app related to enterprise-only features
+  // we don't want to show a user the test treatments if that's the case
+  // since this test doesn't highlight enterprise features at all.
+  const featureFlagValue =
+    commercialFeatureLowestPlan?.[source] !== "enterprise"
+      ? growthbook.getFeatureValue("pro-upgrade-modal", "OFF")
+      : "OFF";
```

Comment:
This logic is a bit interesting so I wanted to call it out.

There are some Upgrade CTAs throughout the app that are referencing Enterprise features. If a user clicks that, this makes it so they see the control treatment (even if they've already be bucketed into the experiment). This is because the two treatments don't have any way of allowing the user to start an enterprise trial. Feels weird to say "upgrade to enterprise" but then only give the "pro" option.

Reply:
makes sense, good catch.  If this version wins, we'll have to make an enterprise-friendly version of it.

`source` is just a string though and is not always a commercial feature.  We should add a new required prop for this modal typed as `CommercialFeature | null` and add it everywhere.  Will be useful for other things as well.

Reply:
We should also only evaluate the feature flag if `isCloud()` and `freeTrialAvailable` are both true.  Otherwise, users may be shown a "start trial" option when they've already used their free trial.  And self-hosted trials require more form inputs to support.

Reply:
@jdorn

Re: the new prop - sounds good - I'll get that added now.

Re: The targeting - I do have that logic built into the experiment rule (default value for the feature flag is "OFF" and then I've got an experiment rule where if the org is cloud, they don't have a licenseKey, and they're in the starter plan, they'll get added to the experiment - is your preference to add that here instead?

Reply:
Any conditions that are strictly required for the code to work should live in the code directly.

Reply:
Sounds good - updated [here](https://github.com/growthbook/growthbook/pull/3650/commits/64fac5732346d8cb0875010e9e7419e080c74cb8).